lib/src/Pos/Launcher/Scenario.hs

```diff
@@ -50,7 +50,7 @@ runNode'
     -> [WorkerSpec m]
     -> [WorkerSpec m]
     -> WorkerSpec m
-runNode' NodeResources {..} workers' plugins' = ActionSpec $ \diffusion -> ntpCheck $ do
+runNode' NodeResources {..} workers' plugins' = ActionSpec $ \diffusion -> maybeNtpCheck $ do
```

Comment:
This is strictly speaking unrelated, but necessary in order to verify this patch. I wanted to run the node while there's no internet connection, but even with the `--no-ntp` flag, the NTP check would be made, and would crash the program concurrently with the DNS subscription thread.

Reply:
https://iohk.myjetbrains.com/youtrack/issue/CSL-2318

Reply:
CSL-2318 is wrong. We always want to do NTP check, unconditionally. `--no-ntp` flag is older than NTP check. It specifies whether we want to use system time to know current time or query NTP servers.
In production we use `--no-ntp`, but we also want NTP check to be enabled.

Reply:
> We always want to do NTP check, unconditionally.

Why? This is relevant to my question on slack: should the node crash if it can't do NTP check? If not, should it block until it can? And a followup: what are the implications for the UI/UX in case it can't do the NTP check?

Reply:
> Why?

I would rather go another way and ask why we may want not to do this check. Probably only because we can't reach NTP servers, but it should be easy to detect it in code.
The reason why `--no-ntp` exists is because NTP-based implementation of slotting (`Pos.Slotting.Impl.Ntp`) doesn't work. Probably it should be removed altogether.
This is a useful check, which is not crucial for the program.
If we can do it, let's do it. If we can't do it, not a problem. There shouldn't be significant overhead in doing this check.

> should the node crash if it can't do NTP check? If not, should it block until it can?

No, it shouldn't. This check is not important. It's mostly for debugging. If it can't do NTP check, blocking until it can is fine.

> And a followup: what are the implications for the UI/UX in case it can't do the NTP check?

There are no implications for UI/UX. `withNtpCheck` is not used by UI. It's used only to print logs. See `onNtpStatusLogWarning` function. Frontend uses `getNtpStatusOnce` function.